src/repository.js

```diff
@@ -39,30 +40,89 @@ module.exports = class Repository {
     return entities
   }
 
+  /** 
+  *
+  * Find all method
+  * 
+  * @param {type}   object.limit Limit items to list  
+  * @param {type}   object.orderBy Order by query
+  *
+  * @return {type} List of entities
+  */
+  async findAll(options = {
+    limit: 0,
+    orderBy: null
+  }) {
+
+    const entities = this.find({ limit: options.limit, orderBy: options.orderBy })
+    return entities
+  }
+
+  /** 
+  *
+  * Find with search term
+  * 
+  * @param {type}   search Term to find entities
+  *
+  * @return {type} Single entity
+  */
   async findBy(search) {
+    const entities = await this.find({ search });
+    return entities;
+  }
+
+  /** 
+  *
+  * Find entities
+  * 
+  * @param {type}   object.limit Limit items to list  
+  * @param {type}   object.offset Offset records
+  * @param {type}   object.search Where query term
+  * @param {type}   object.orderBy Order by query
+  *
+  * @return {type} List of entities
+  */
+  async find(options = {
+    limit: 0,
+    offset,
+    orderBy: null,
+    search: null
```

Comment:
I think `conditions` is a better name instead of `search`

https://www.postgresql.org/docs/current/sql-select.html

Reply:
Done!